{"cells":[{"metadata":{"_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19","trusted":true},"cell_type":"code","source":"# This Python 3 environment comes with many helpful analytics libraries installed\n# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python\n# For example, here's several helpful packages to load\n\nimport numpy as np # linear algebra\nimport pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\n\n# Input data files are available in the read-only \"../input/\" directory\n# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory\n\nimport os\nfor dirname, _, filenames in os.walk('/kaggle/input'):\n    for filename in filenames:\n        print(os.path.join(dirname, filename))\n\n# You can write up to 5GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using \"Save & Run All\" \n# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session","execution_count":null,"outputs":[]},{"metadata":{"_uuid":"d629ff2d2480ee46fbb7e2d37f6b5fab8052498a","_cell_guid":"79c7e3d0-c299-4dcb-8224-4455121ee9b0","trusted":true},"cell_type":"code","source":"import pandas as pd\nimport numpy as np\nimport  re\npd.set_option('display.max_columns', 50)","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#Reading in Model List\nmodel_list = pd.read_csv(\"../input/model-list/model_list_20200825 (1).csv\")\n#Cell Line Mapping\ncell_line_mapping = model_list[['model_id','sample_id','model_name','COSMIC_ID','BROAD_ID','CCLE_ID','tissue','tissue_status','cancer_type']].copy()\n#cell_line_mapping.shape\ncell_line_mapping.head()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#Reading in Gene Mapping\ngene_mapping= pd.read_csv(r'../input/gene-mapping-updated/Gene Mapping_20191101.csv')\nprint(gene_mapping.shape)\ngene_mapping.head()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#readings metabolomics data\nccle_metabolomics_original = pd.read_csv(r'../input/ccle-metabolomics/CCLE_metabolomics_20190502 (3).csv')\nccle_metabolomics = ccle_metabolomics_original.copy()\nprint(ccle_metabolomics.shape)\nccle_metabolomics.head()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"# map CCLE_ID to DepMap_ID\nccle_metabolomics = ccle_metabolomics.merge(cell_line_mapping[['CCLE_ID','model_id']],\n                     left_on = 'CCLE_ID',\n                     right_on = 'CCLE_ID',\n                     how = 'left')\nprint('percentage of na in model_id: ', ccle_metabolomics['model_id'].isna().sum()/ccle_metabolomics.shape[0])\nnan_count = ccle_metabolomics['model_id'].isna().sum()\nprint(nan_count)\n\nprint(ccle_metabolomics.shape)\n\n#Drop missing values in column 'model_id'\nccle_metabolomics.dropna(subset=['model_id'], inplace=True) \n\nprint(ccle_metabolomics.shape)\nccle_metabolomics.head()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"#Remove CCLE_ID & DepMap_ID\nmatrix_metabolomics = ccle_metabolomics.iloc[:,2:]\nmatrix_metabolomics = matrix_metabolomics[[list(matrix_metabolomics.columns)[-1]]+ list(matrix_metabolomics.columns[:-1])]\n\nmatrix_metabolomics.to_csv('./matrix_metabolomics.csv', index = False)\nmatrix_metabolomics.to_csv('./matrix_metabolomics_pca.csv', header=False, index = False)\n\nprint(matrix_metabolomics.shape)\nmatrix_metabolomics.head()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"ccle_rppa_original = pd.read_csv(r'../input/ccle-rppa/CCLE_RPPA_20181003 (3).csv')\nccle_rppa = ccle_rppa_original.copy()\nprint(ccle_rppa.shape)\n\n# map CCLE_ID to model_id\nccle_rppa.rename(columns={'Unnamed: 0':'CCLE_ID'},inplace=True)\nmatrix_rppa = ccle_rppa.merge(cell_line_mapping[['CCLE_ID','model_id']],\n                     left_on = 'CCLE_ID',\n                     right_on = 'CCLE_ID',\n                     how = 'left')\n#Removing NA values (meaning no model ID matching)\n\nprint('percentage of na in model_id: ', matrix_rppa['model_id'].isna().sum()/matrix_rppa.shape[0])\nnan_count = matrix_rppa['model_id'].isna().sum()\nprint(nan_count)\n\nprint(matrix_rppa.shape)\nmatrix_rppa.dropna(subset=['model_id'], inplace=True)\nprint(matrix_rppa.shape)\n\nmatrix_rppa = matrix_rppa.iloc[:,1:]\nmatrix_rppa = matrix_rppa[[list(matrix_rppa.columns)[-1]]+ list(matrix_rppa.columns[:-1])]\nmatrix_rppa.to_csv('./matrix_rppa.csv', index = False)\nmatrix_rppa.to_csv('./matrix_rppa_pca.csv', header=False, index = False)\n\n# In[23]:\n\n\nprint(matrix_rppa.shape)\nmatrix_rppa.head()\n\n\n# In[24]:\n\n\n#ccle_rppa_original.head()\n\n\n","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"gdsc_cnv_original = pd.read_csv(r'../input/gene-cnv-2018-1113-1545/gene_cnv_2018-11-13_1545.csv')\ngdsc_cnv = gdsc_cnv_original.copy()\nprint(gdsc_cnv.shape)\n\nmatrix_cnv = gdsc_cnv[['model_id','gene_id','abs_copy_number']]\nprint(matrix_cnv.shape)\n\nmatrix_cnv.drop_duplicates(inplace = True)\nprint(matrix_cnv.shape)\n\nmatrix_cnv.dropna(inplace = True)\nprint(matrix_cnv.shape)\nmatrix_cnv = matrix_cnv.pivot_table(index = 'model_id',\n                          columns = 'gene_id',\n                        values = 'abs_copy_number')\nprint(matrix_cnv.shape)\nmatrix_cnv.to_csv('./matrix_cnv.csv', index = True)\nmatrix_cnv.to_csv('./matrix_cnv_pca.csv', header=False, index = False)\n\n\n#To count no. of NAN in df\n# print('NAN values',matrix_cnv.isnull().sum().sum())\n#gdsc_cnv_original.head()\n\nmatrix_cnv.head()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"ccle_expression_original = pd.read_csv(r'../input/ccle-expression-0920/CCLE_expression_ 0920.csv')\nccle_expression = ccle_expression_original.copy()\nprint(ccle_expression.shape)\nccle_expression.head()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"# map BROAD_ID to model_id\nccle_expression.rename(columns={'Unnamed: 0':'BROAD_ID'},inplace=True)\nccle_expression = ccle_expression.merge(cell_line_mapping[['BROAD_ID','model_id']],\n                     left_on = 'BROAD_ID',\n                     right_on = 'BROAD_ID',\n                     how = 'left')\n\nprint(ccle_expression.shape)\n\nnan_count = ccle_expression['model_id'].isna().sum()\nprint(nan_count)\n\nccle_expression.dropna(subset=['model_id'], inplace=True)\nprint(ccle_expression.shape)\n\nmatrix_expression = ccle_expression.iloc[:,1:]\nmatrix_expression = matrix_expression[[list(matrix_expression.columns)[-1]]+ list(matrix_expression.columns[:-1]) ]","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"# map cosmic_gene_symbol to gene_id\nmatrix_expression_t = matrix_expression.T\nmatrix_expression_t.reset_index(level=0, inplace=True)\nmatrix_expression_t['index'] = matrix_expression_t['index'].str.split(' ', expand = True)[0]\nmatrix_expression_t = matrix_expression_t.merge(gene_mapping[['cosmic_gene_symbol','gene_id']],\n                     left_on = 'index',\n                     right_on = 'cosmic_gene_symbol',\n                     how = 'left')\n\nmatrix_expression_t['gene_id'][0] = 'model_id'\n\nmatrix_expression.columns = matrix_expression_t['gene_id']\nmatrix_expression = matrix_expression.iloc[:,matrix_expression.columns.notna()]\nmatrix_expression.to_csv('./matrix_expression.csv', index = False)\nmatrix_expression.to_csv('./matrix_expression_pca.csv', header=False, index = False)","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"print(matrix_expression.shape)\nmatrix_expression.head()\n\n#ccle_expression.head()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"ccle_mutations_original = pd.read_csv(r'../input/ccle-mutations-comma-delimited/CCLE_mutations_comma_delimited.csv')\nccle_mutations = ccle_mutations_original.copy()\nprint(ccle_mutations.shape)\nccle_mutations.head()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"# simplify variant classification\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification']\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Missense_Mutation\",\"Missense\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Silent\",\"Silent\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Splice_Site\",\"Splice_Site\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Frame_Shift_Del\",\"Deletion\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Nonsense_Mutation\",\"Nonsense\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Frame_Shift_Ins\",\"Insertion\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"In_Frame_Del\",\"Deletion\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"De_novo_Start_OutOfFrame\",\"Other\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"In_Frame_Ins\",\"Insertion\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Start_Codon_SNP\",\"Other\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Nonstop_Mutation\",\"Nonstop\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Intron\",\"Other\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Stop_Codon_Del\",\"Deletion\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Stop_Codon_Ins\",\"Insertion\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Start_Codon_Del\",\"Deletion\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"IGR\",\"Other\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"Start_Codon_Ins\",\"Insertion\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"5'Flank\",\"Other\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"3'UTR\",\"Other\")\nccle_mutations['Variant_Classification_Simplified'] = ccle_mutations['Variant_Classification_Simplified'].str.replace(\"5'UTR\",\"Other\")\n\n","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"# map DepMap_ID to model_id\nccle_mutations = ccle_mutations.merge(cell_line_mapping[['BROAD_ID','model_id']],\n                     left_on = 'DepMap_ID',\n                     right_on = 'BROAD_ID',\n                     how = 'left')\n\n# map Entrez_Gene_Id to gene_id\nccle_mutations = ccle_mutations.merge(gene_mapping[['entrez_id','gene_id']],\n                     left_on = 'Entrez_Gene_Id',\n                     right_on = 'entrez_id',\n                     how = 'left')\n\n# remove duplicates\nprint(ccle_mutations.shape)\nccle_mutations.drop_duplicates(subset=['gene_id','model_id','Variant_Classification_Simplified'], inplace=True)\nprint(ccle_mutations.shape)\n\n\nprint('percentage of na in model_id: ', ccle_mutations['model_id'].isna().sum()/ccle_mutations.shape[0])\nprint('percentage of na in gene_id: ', ccle_mutations['gene_id'].isna().sum()/ccle_mutations.shape[0])\n","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"matrix_mutations = ccle_mutations.pivot_table(index = 'model_id',\n                          columns = 'gene_id',\n                        values = 'Variant_Classification_Simplified',\n                        aggfunc='first')\n\nmatrix_mutations_binary = matrix_mutations.copy()\nprint(matrix_mutations.shape)\n\nmatrix_mutations[matrix_mutations.isna()] = 0  \nmatrix_mutations_binary[matrix_mutations_binary.notna()] = 1  \nmatrix_mutations_binary[matrix_mutations_binary.isna()] = 0  \n\n\n\n\nmatrix_mutations_frequency = ccle_mutations.pivot_table(index = 'model_id',\n                          columns = 'gene_id',\n                        values = 'Variant_Classification_Simplified',\n                        aggfunc='count')\nmatrix_mutations_frequency[matrix_mutations_frequency.isna()] = 0  \nprint(matrix_mutations.shape)\n\nmatrix_mutations.head(10)\n\n\n\nmatrix_mutations.to_csv('./matrix_mutations.csv', index = True)\nmatrix_mutations_binary.to_csv('./matrix_mutations_binary.csv', index = True)\nmatrix_mutations_binary.to_csv('./matrix_mutations_binary_pca.csv', header=False, index = False)","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"gdsc_drugResponse = pd.read_excel(r'../input/gdsc2-fitted-dose-response-25feb20/GDSC2_fitted_dose_response_25Feb20.xlsx')\nprint(gdsc_drugResponse.shape)\ngdsc_drugResponse.head()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"gdsc_drugResponse.rename(columns = {'SANGER_MODEL_ID':'model_id', 'DRUG_ID':'drug_id'}, inplace=True)\n\nmatrix_drug_response = gdsc_drugResponse.pivot_table(columns='drug_id',\n                             index='model_id',\n                             values='LN_IC50')\nmatrix_drug_response[matrix_drug_response.isna()] = 0  \n\nmatrix_drug_response.head()\nmatrix_drug_response.shape","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"matrix_drug_response.to_csv('./matrix_drug_response.csv', index=True)\n","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"ccle_miRNA_original = pd.read_csv(r'../input/cclemirna/CCLE_miRNA_20181103.csv')\nccle_miRNA = ccle_miRNA_original.copy()\n# print(ccle_miRNA.shape)\nccle_miRNA.head()\n","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"miRNA_t = ccle_miRNA.T\nprint(miRNA_t.shape)","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"miRNA_t.head()\nmiRNA_t.to_csv('./miRNA_t.csv', header=False, index = False)","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"miRNA_trans = pd.read_csv(r'../input/mirna-trans/miRNA_trans.csv')\nmiRNA_trans.head()","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"# map CCLE_ID to model_id\nmiRNA_trans.rename(columns={'Unnamed: 0':'CCLE_ID'},inplace=True)\nmatrix_mirRNA = miRNA_trans.merge(cell_line_mapping[['CCLE_ID','model_id']],\n                     left_on = 'CCLE_ID',\n                     right_on = 'CCLE_ID',\n                     how = 'left')\n#Removing NA values (meaning no model ID matching)\n\nprint('percentage of na in model_id: ', matrix_mirRNA['model_id'].isna().sum()/matrix_mirRNA.shape[0])\nprint(matrix_mirRNA.shape)\n\nmatrix_mirRNA.dropna(subset=['model_id'], inplace=True)\nmatrix_mirRNA = matrix_mirRNA.iloc[:,1:]\nmatrix_mirRNA = matrix_mirRNA[[list(matrix_mirRNA.columns)[-1]]+ list(matrix_mirRNA.columns[:-1])]\nmatrix_mirRNA.to_csv('./matrix_mirRNA.csv', index = False)\nmatrix_mirRNA.to_csv('./matrix_mirRNA_pca.csv', header=False, index = False)\n\n# In[23]:\n\n\nprint(matrix_mirRNA.shape)\nmatrix_mirRNA.head()\n","execution_count":null,"outputs":[]}],"metadata":{"kernelspec":{"name":"python3","display_name":"Python 3","language":"python"},"language_info":{"pygments_lexer":"ipython3","nbconvert_exporter":"python","version":"3.6.4","file_extension":".py","codemirror_mode":{"name":"ipython","version":3},"name":"python","mimetype":"text/x-python"}},"nbformat":4,"nbformat_minor":4}